### PR TITLE
Fix for PR#73 and minor clean up

### PR DESCRIPTION
**Pull request by bvavala**
-conditional docker command executions
-force file rm
-typos in commands
It addresses #71

### CODE REVIEW

File: tlcc_enclave/generate_mrenclave.sh [hunk 1]
```diff
@@ -13,7 +13,7 @@ tmp_name=tmp_enc_hash
 go_name=mrenclave.go
 
 cd $build_dir
-sgx_sign gendata -enclave enclave.so -config $enclave_dir/Enclave.config.xml -out $tmp_name
+sgx_sign gendata -enclave enclave.so -config $enclave_dir/enclave.config.xml -out $tmp_name
```
Comment (g2flyer) [inline tlcc_enclave/generate_mrenclave.sh]: Good catch, but the bigger problem seems that we do no catch any errors and so it was happily running so far (just producing only an empty mrenclave).   Easiest fix for that seems to be o just add a 'set -e' in the beginning?

Also, given that the generate_mrenclave.sh is the same in ecc_enclave and tlcc_enclave, maybe we should move that to a common bin directory? 

Thinking of it, the method here also seems rather brittle, the way we do it in pdo is more robust:
- add to sgx_sign sign the '-dumpfile "${SIGNED_ENCLAVE_METADATA}"' option and then
- extract it with a more understandable like the ' VAR_MRENCLAVE=$$(perl -0777 -ne 'if (/metadata->enclave_css.body.enclave_hash.m:([a-fx0-9 \n]+)/) { $$eh = $$1; $$eh=~s/0x| |\n//g; $$eh=~tr/a-z/A-Z/; print "$${eh}\n"; }' $(CONTRACT_ENCLAVE_MRENCLAVE_META_FILE))' in target '$(CONTRACT_ENCLAVE_MRENCLAVE_C_FILE)' in pservice/Makefile

Comment (mbrandenburger) [reply tlcc_enclave/generate_mrenclave.sh]: This deserves a separate PR! I created an issue for that #77

File: ecc/Makefile [hunk 2]
```diff
@@ -72,9 +78,9 @@ docker-run:
 		chaincode
 
 docker-stop:
-	-docker stop $(DOCKER_CONTAINER)
-	-docker rm $(DOCKER_CONTAINER)
+	-if [ ! -z "$(DOCKER_CONTAINER_ID)" ]; then docker stop $(DOCKER_CONTAINER); fi
```
Comment (mbrandenburger) [inline ecc/Makefile]: this should be `docker stop $(DOCKER_CONTAINER_ID);`, right?

also DOCKER_CONTAINER is not really useful anymore, I think DOCKER_CONTAINER_ID is much better.

I suggest let's remove DOCKER_CONTAINER and in docker-run target we just use CC_NAME.

File: ecc/Makefile [hunk 3]
```diff
@@ -56,6 +58,10 @@ clean: docker-clean
 #	rm enclave/mrenclave.go
 
 docker:
+	@if [ -z "${DOCKER_IMAGE}" ]; then\
+		echo "\033[0;31mERROR: no $(CC_NAME) docker image found\033[0m";\
```
Comment (mbrandenburger) [inline ecc/Makefile]: This error message seems to be not super intuitive. As `make docker` creates a docker image for CC_NAME, the error tells the user that here is no such image. mhhh.

I suggest the following.

If there is no ercc docker image yet, output a warning (maybe in yellow if you want) and create the image with some default name which can be set by the user.

If the docker image exists, maybe also output a warning that we override an existing image.

WDYT?

File: ecc/Makefile (hunk 2 above)
Comment (bvavala) [reply ecc/Makefile]: agreed

File: ecc/Makefile (hunk 3 above)
Comment (bvavala) [reply ecc/Makefile]: I changed the message.
For the rest I am not sure what the "current" use of that would be. I would postpone it to a separate PR that addresses this.

Comment (mbrandenburger) [reply ecc/Makefile]: ok for me

**Comment by mbrandenburger** [comment]
Thanks Bruno!

**Pull request merged**